internal/controller/services/monitoring/monitoring_controller_actions.go

```diff
@@ -109,12 +114,52 @@ func createMonitoringStack(ctx context.Context, rr *odhtypes.ReconciliationReque
 	if !msExists {
 		return errors.New("MonitoringStack CRD not found")
 	}
+
 	if monitoring.Spec.Metrics != nil {
-		template := []odhtypes.TemplateInfo{
+		templates := []odhtypes.TemplateInfo{
 			{
 				FS:   resourcesFS,
 				Path: MonitoringStackTemplate,
 			},
+			{
+				FS:   resourcesFS,
+				Path: PrometheusRouteTemplate,
+			},
+		}
+
+		rr.Templates = append(rr.Templates, templates...)
+	}
+
+	return nil
+}
+
+func createOpenTelemetryCollector(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	otcExists, _ := cluster.HasCRD(ctx, rr.Client, gvk.OpenTelemetryCollector)
```

Comment:
maybe we should also add a watch on the OpenTelemetryCollector CRD, so when the CRD gets installed, the reconcile would get triggered

Reply:
CRD already triggers reconciliation via: https://github.com/opendatahub-io/opendatahub-operator/blob/1c97b84a326b1ded446109353045624317a3afb1/internal/controller/services/monitoring/monitoring_controller.go#L97-L101